Assets/PatchKit Patcher/Scripts/AppUpdater/Commands/CheckVersionIntegrityCommand.cs

```diff
@@ -95,19 +97,80 @@ public override void Execute(CancellationToken cancellationToken)
             }
         }
 
+        private static volatile FileIntegrity[] files;
+
         private void ExecuteInternal(CancellationToken cancellationToken)
         {
             DebugLogger.Log("Checking version integrity.");
 
             _status.IsActive.Value = true;
 
-            var files = new FileIntegrity[_versionSummary.Files.Length];
-
-            for (int i = 0; i < _versionSummary.Files.Length; i++)
+            files = new FileIntegrity[_versionSummary.Files.Length];
+            int i = 0;
+            foreach (var file in _versionSummary.Files)
```

Comment:
`for` can be reverted here, so only `i` needs to be passed to the thread (`_versionSummary.Files` is not modified anywhere so can be freely accessed from worker thread).

Reply:
@genail you suggested using `foreach` here because it works faster.